Add Ctrl+B keyboard shortcut to toggle sidebar visibility

arturhg opened the pull request:
## Summary
This PR adds a keyboard shortcut (Ctrl+B / Cmd+B on Mac) to quickly toggle the visibility of both sidebars, addressing issue #206.

## Changes
- Implemented `toggleSideBarsVisibility()` static method in `SideBar.java` to toggle both sidebars
- Added `toggleVisibility()` instance method to handle individual sidebar toggling with proper state tracking
- Registered Ctrl+B (Cmd+B on Mac) shortcut in `KeyboardShortcuts.java`
- Added translation keys for the new shortcut in:
  - English: "Toggle visibility of sidebars"
  - French: "Afficher/masquer les panneaux latéraux"
  - Italian: "Mostra/nascondi i pannelli laterali"

## Notes
- The shortcut uses the platform-appropriate modifier (Ctrl on Windows/Linux, Cmd on Mac)
- Empty sidebars (no tabs) are not toggled to avoid confusion
- Documentation (ODT and PDF files) will need to be updated to include this new shortcut

Fixes #206

ClementGre commented:
There are some bugs with your implementation.
`removeToPane()` and `addToPane()` are made to support having one sidebar, two sidebars, but never 0 sidebars. In this case, it throws the following exception if trying to close sidebars while using only the left bar. An analogue error occurs if trying to open sidebars after closing with both left and right sidebars opened:
```
java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at fr.clementgre.pdf4teachers/fr.clementgre.pdf4teachers.panel.sidebar.SideBar.getWidthByDivider(SideBar.java:178)
	at fr.clementgre.pdf4teachers/fr.clementgre.pdf4teachers.panel.sidebar.SideBar.removeToPane(SideBar.java:202)
	at fr.clementgre.pdf4teachers/fr.clementgre.pdf4teachers.panel.sidebar.SideBar.toggleVisibility(SideBar.java:305)
```

You could fix it by editing `getWidthByDivider()`. But it might be better to, instead of removing the sidebars, set their width to 0 so the user can still reopen them with the mouse and will never get stuck if they forget the shortcut or hit it accidentally.

When using only a right sidebar. Closing it and then reopening it works (the only configuration that works all the way for me), but the restored sidebar does not get its original width and takes the maximum sidebar width.

Last but not least, I think it might be better to put the shortcut in the menubar. PDF4Teachers does not have a View menu, so I don’t know what to do. Maybe we could create a new View menu and add other menu items such as "Restore default sidebar configuration", "Move all tabs to the right(or left)", the "minimize sidebars" item, and any other useful view action.
What do you think about this?

ClementGre commented:
Ok, that's a good start, but there are still some issues:
- When moving tabs to the left or right, it sets the size of the emptied sidebar to 0 instead of removing it from the pane.
- The keyboard shortcut is not working on MacOS (I think you forgot to remove the entry in KeyboardShortcuts.java)
- The minimize sidebar option is not very useful I think. What usage do you see in this feature ? 
- I think we could move the full screen mode menu item from Tools to View.
- It would be great to make the view menu dynamic (though it might not work on macos, I can check the implementation later if you are not on MacOS). What I mean by that is removing or just disabling the items that have no effect in the current disposition of the sidebar (disable or remove the item move tabs to left if they are already at the left, etc.). You can look at items Undo/Redo to see how reactivity is implemented in the menu bar. It could also be great to implement shortcuts for the toggle fullscreen item (at least on MacOS : cmd + ctrl + F), and maybe make it reactive (enter full screen/quit full screen).

I honestly feel like I’m talking to a LLM, so if you are not into testing the implementation, challenging and reviewing yourself (or your LLM), I think I can do it myself; it would be faster to remove a level of indirection.

arturhg commented:
Thank you for the feedback. Let me address each point:

You are not talking to an LLM, but I do use LLMs to help express my thoughts more clearly since English is not my native language and I'm far from fluent. If this bothers you to the point of being irritating, please let me know directly and I will stop submitting PRs to the project.

I'm testing on Linux and have very limited access to macOS - mostly having to ask others to test things for me occasionally. This makes it difficult to catch platform-specific issues like the keyboard shortcut not working on macOS.

> When moving tabs to the left or right, it sets the size of the emptied sidebar to 0 instead of removing it from the pane.

Empty sidebars are now completely removed from the pane when all tabs are moved to the other side.

> On Mac OS, the keyboard shortcut doesn't work now. Remove the duplicate, keep only the menu item entry.

Removed the duplicate shortcut from KeyboardShortcuts.java, keeping only the menu item entry.

> The minimize sidebar option is not very useful I think. What usage do you see in this feature ?

You're right, it's not very useful in its current form. The idea was that even when minimized, it would still show some information, but the current implementation doesn't achieve this. I've removed it and it could potentially be revisited in the future with a proper design.

> I think we could move the full screen mode menu item from Tools to View.

Moved fullscreen from Tools menu to View menu.

> Could you add the Cmd+Ctrl+F keyboard shortcut on Mac OS for the full-screen mode?

Added platform-specific keyboard shortcut for fullscreen: Cmd+Ctrl+F on macOS, F11 on other platforms. The shortcut is now correctly set based on the platform using `PlatformUtils.isMac()`.

> Also try to make the menu items more reactive (eg. disabling the move tabs to left item when all tabs are already on the left). You can look at the implementation of the Undo/Redo menu items.

Implemented reactive menu items using the ON_SHOWING event handler. The "Move all tabs to left" option is now disabled when all tabs are already on the left, and "Move all tabs to right" is disabled when all tabs are on the right.

> Make the fullscreen menu item update its text dynamically ("Enter Full Screen", "Exit Full Screen") 

The fullscreen menu item now dynamically updates its text between "Enter Full Screen" and "Exit Full Screen" based on the current fullscreen state. Also cleaned up old Tools menu translations for fullscreen since it's now in the View menu.

ClementGre commented:
Ok, thank you for the clarification.
I like what you did, but there are still some changes and fixes that might be worth it.

- In removeToPane(), you set the max width to 1px, but this prevents the user from reopening it manually. The idea of making the grab bar visible is for the user to be able to reopen the tab in case of mistyping Ctrl+B and not knowing the shortcut. Then setting the bar width to 1px without changing its max width should do the job. Then the toggle sidebars would set the default width to sidebars that are 1px wide (or 0px wide), and if no sidebar is that small, it should set to 1px/0px all sidebars.
- When moving tabs to the left or right right after opening the app, I have the new sidebar taking its maximum width. You should set its width to the default width each time the sidebar reopens (which was not the original behaviour btw).
- If you want, it could be cool to save the "user sidebar width" in the `SyncUserData`, then, the default width would only be used as the default value of the sidebar width in `SyncUserData`, and the user sidebar width would always be correctly restored. If the user closes a sidebar by dragging it manually, it would either keep the user sidebar width as the last opened value, or it would set it to 0, and bring it back to the default width if pressing ctrl+B.
- Maybe we could replace the menu item "Restore default sidebar width" to "Restore default sidebars", moving tabs to the left, reordering tabs, and setting the default sidebar width.
